packages/republik/lib/RevenueStats/segments.js

```diff
@@ -0,0 +1,148 @@
+const debug = require('debug')('republik:lib:RevenueStats:segments')
+
+const {
+cache: { create },
+} = require('@orbiting/backend-modules-utils')
+
+const QUERY_CACHE_TTL_SECONDS = 60 * 60 * 24 * 8 // A week and a day
+const BUCKET_DATE_FORMAT = 'YYYY'
+
+const query = `
+WITH "totals" AS (
+  WITH "resolvedData" AS (
+    WITH "packageOptionsWithRewardType" AS (
+      SELECT pkgo.id, r.type
+      FROM "packageOptions" pkgo
+      JOIN "rewards" r ON r.id = pkgo."rewardId"
+    )
+    SELECT
+      pay."createdAt" "paymentCreatedAt",
+      pay.id "paymentId",
+      pay.total "paymentTotal",
+
+      -- Memberships
+      date(p."createdAt") != date(ppay."createdAt") "autoPay",
```

Comment:
```suggestion
      p."createdAt"::date != ppay."createdAt"::date "autoPay",
```

(Accustomed to casting date via `::`)